apps/src/blockly/addons/cdoUtils.js

```diff
@@ -2,9 +2,26 @@ import {ToolboxType, CLAMPED_NUMBER_REGEX, DEFAULT_SOUND} from '../constants';
 import cdoTheme from '../themes/cdoTheme';
 import {APP_HEIGHT} from '@cdo/apps/p5lab/constants';
 import {SOUND_PREFIX} from '@cdo/apps/assetManagement/assetPrefix';
+import {convertXmlToJson, positionBlocks} from './cdoSerializationHelpers';
+import experiments from '@cdo/apps/util/experiments';
 
+/**
+ * Loads blocks to a workspace.
+ * To maintain backwards compatibility we must be able to use the XML source if no JSON state is provided.
+ * @param {Blockly.Workspace} workspace - the current Blockly workspace
+ * @param {xml} xml - workspace serialization, current/legacy format
+ * @param {*} stateToLoad - modern workspace serialization, may not be present
+ */
 export function loadBlocksToWorkspace(workspace, xml, stateToLoad) {
```

Comment:
Is this list of arguments copied from google blockly, or is it our own? I'm a little confused at the names "xml" and "stateToLoad"...is `stateToLoad` the same thing as "xml" just in a json format?

Reply:
This function is our own as are the parameters.  Throughout Google Blockly core, they use `state` to refer to the JSON-formatted serialization that is being loaded. But yes, both `xml` and `stateToLoad` are different formats of the same thing. Open to suggestions for naming here!

Reply:
Ok cool if it's consistent with them, good with me. Do they use `xml` to refer to what we call `xml` here? 

We could align the naming in this function (`xmlState`, `jsonState`) if we want to make clear to future engineers that it's the same thing...whatever you think is better.

Reply:
I like that suggestion (xmlState, jsonState).

And yes, like us, they use `xml` for the older XML functions.